contracts/BaseWrapper.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {RegistryAPI, VaultAPI} from "./BaseStrategy.sol";
+
+abstract contract BaseWrapper {
+    using SafeMath for uint256;
+
+    ERC20 public token;
+
+    // Reduce number of external calls (SLOADs stay the same)
+    VaultAPI[] private _cachedVaults;
+
+    RegistryAPI public registry;
+
+    constructor(address _token) public {
+        token = ERC20(_token);
+        // v2.registry.ychad.eth
+        registry = RegistryAPI(0xE15461B18EE31b7379019Dc523231C57d1Cbc18c);
```

Comment:
i guess we can check if token has a latestDEployment or blow up here 

`require(registry.latestVault(address(token) != address(0))`
?

Reply:
`latestVault` reverts if there's no latest

Reply:
yeah thinking we revert at contract deployment to not have an affiliate token with a non supported token/vault